core/ArchiveProcessor/Rules.php

```diff
@@ -328,17 +328,20 @@ public static function getSelectableDoneFlagValues($includeInvalidated = true, P
         return $possibleValues;
     }
 
-    public static function isForceArchivingSinglePlugin()
+    public static function isRequestingToAndAbleToForceArchiveSinglePlugin()
     {
-        if (!SettingsServer::isArchivePhpTriggered()
-            || Loader::getArchivingDepth() > 1
-        ) {
+        if (!SettingsServer::isArchivePhpTriggered()) {
             return false;
         }
 
         return !empty($_GET['pluginOnly']) || !empty($_POST['pluginOnly']);
     }
 
+    public static function isActuallyForceArchivingSinglePlugin()
+    {
+        return Loader::getArchivingDepth() <= 1 && self::isRequestingToAndAbleToForceArchiveSinglePlugin();
+    }
```

Comment:
We now have two methods to determine if we are forcing archiving of a single plugin:

- isRequestingToAndAbleToForceArchiveSinglePlugin() which returns true if the original request was for a pluginOnly archive, and if it came from archive.php (so we're able to in that case).
- and isActuallyForceArchivingSinglePlugin() which returns true if in this specific archiving request, we want to archive a pluginOnly archive. We only do this for the root request, for others we try to reuse existing archives. If no existing archives exist, archiving will launch for them so we don't need to do anything (unless, like in the Cloud plugin, we force disable it).